Mobile Buy SDK/Mobile Buy SDK/Data/BUYClient+Checkout.m

```diff
@@ -266,11 +218,68 @@ - (BUYRequestOperation *)storeCreditCard:(BUYCreditCard *)creditCard checkout:(B
 	}];
 }
 
-- (BUYRequestOperation *)removeProductReservationsFromCheckout:(BUYCheckout *)checkout completion:(BUYDataCheckoutBlock)block
+- (BUYRequestOperation *)applyGiftCardCode:(NSString *)giftCardCode toCheckout:(BUYCheckout *)checkout completion:(BUYDataCheckoutBlock)block
+{
+	BUYAssertCheckout(checkout);
+	BUYAssert(giftCardCode.length > 0, @"Failed to apply gift card code. Invalid gift card code.");
+	
+	BUYGiftCard *giftCard = [self.modelManager giftCardWithCode:giftCardCode];
+	NSURL *route = [self urlForCheckoutsUsingGiftCardForToken:checkout.token];
+	
+	return [self postRequestForURL:route object:giftCard completionHandler:^(NSDictionary *json, NSHTTPURLResponse *response, NSError *error) {
+		if (json && !error) {
+			
+			NSDictionary *giftCardJSON = json[@"gift_card"];
+			BUYGiftCard *giftCard      = [self.modelManager insertGiftCardWithJSONDictionary:giftCardJSON];
+			
+			/* ---------------------------------
+			 * We have to update a copy of the
```

Comment:
As part of a lengthy discussion with @bgulanowski, we decided not to copy the checkout. The comments were not updated. Will fix.